armi/reactor/geometry.py

```diff
@@ -67,21 +70,22 @@ def fromAny(cls, source: Union[str, "GeomType"]) -> "GeomType":
             raise TypeError("Expected str or GeomType; got {}".format(type(source)))
 
     @classmethod
-    def fromStr(cls, geomStr):
+    def fromStr(cls, geomStr: str) -> "GeomType":
         # case-insensitive
-        canonical = geomStr.lower().strip()
-        if canonical == HEX or canonical == HEX_CORNERS_UP:
-            # corners-up is used to rotate grids, but shouldn't be needed after the grid
-            # is appropriately oriented, so we collapse to HEX in the enumeration. If
-            # there is a good reason to make corners-up HEX its own geom type, we will
-            # need to figure out how to design around that.
-            return cls.HEX
-        elif canonical == CARTESIAN:
-            return cls.CARTESIAN
-        elif canonical == RZT:
-            return cls.RZT
-        elif canonical == RZ:
-            return cls.RZ
+        x = geomStr.lower().strip()
+        for canonical in (x, parsing.findString(geomTypes, x)):
```

Comment:
The idea was that if you pass a string that has a `GeomType` in it, but doesn't exactly match a `GeomType`, this would still return the type that is in the string. For example, if you passed `"hexagon"`, findString() finds the string `"hex`" inside of `"hexagon"`, and then it still returns `"geometry.GeomType.HEX"`.

I suppose it could also get confused though, because if you passed `"rz theta"` it would find `"rz"`, not `"thetarz"`.

I did it first for symmetry, so you could find `"third"` inside of `"third periodic"`, and then thought it might be helpful for `GeomType` also. But it might not be doing much here.

Reply:
Yeah, i tend to lean towards being more pedantic than permissive in input processing, if it should be "hex," it should be "hex" unless there is a really good reason. for the more complicated case of the 1/Nth core plus boundary condition, i would still handle that somewhat explicitly; split on space, zero-th element goes to the domain parser bit, last element to the BC, special case for the "though center" stuff.

Reply:
I'll take it out for the GeomType.

For the more complicated case, I think there could still be value in this approach, because you can construct a ShapeType or a BoundaryType independently of a SymmetryType by passing the whole string without pre-parsing it. But I don't know how frequently we would want to do that.

Reply:
The counter-example to being very strict about the input options is that the phrase `"third core periodic"` and `"eighth core periodic"` are all over the place in the repository, while `"third periodic"` is technically the correct form. We could handle that in the parsing, but we're just allowing one specific type of ambiguity.

Reply:
If we expect/allow "core" to be thrown in there fore the user's comfort, then we can always choose to scrub that out. I just don't know if it makes sense to open the door to parsing
```
Lorem ipsum dolor *eighth* amet, consectetur adipiscing elit, sed
do eiusmod tempor incididunt ut labore et dolore magna aliqua.
Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris
nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in
reprehenderit in voluptate velit *core* cillum dolore eu fugiat nulla
pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
culpa qui officia deserunt mollit anim id est laborum.
```

Reply:
I agree there should be some limit on it. Allowing arbitrary random input was just a side effect; my initial goal was just to find the relevant strings with the simplest code possible. It was easier to code "`if x in y`" than to go with a `y.split()', where the `split` can yield 1, 2, 3, 5, or 6 elements, and having to explicitly match up variable with element in each case. But the implementation with splitting isn't that bad.

As for the strings in the code, I was mostly referring to `*.xml`` input files in the repository used for testing. I think I replaced all of the strings in actual `*.py` with enumerations.